aten/src/ATen/functorch/BatchRulesScatterOps.cpp

```diff
@@ -1056,6 +1056,150 @@ std::tuple<Tensor,optional<int64_t>> masked_fill_scalar_batch_rule(
   return std::make_tuple(result, 0);
 }
 
+std::tuple<Tensor,optional<int64_t>> index_fill_int_scalar_batch_rule_impl(
+    Tensor & self, optional<int64_t> self_bdim,
+    int64_t dim,
+    const Tensor & index, optional<int64_t> index_bdim,
+    const Scalar & value,
+    const bool inplace) {
+
+    Tensor self_ = moveBatchDimToFront(self, self_bdim);
+    Tensor index_ = moveBatchDimToFront(index, index_bdim);
+    dim = maybe_wrap_dim(dim, self_.dim());
```

Comment:
This should be
```
dim =  maybe_wrap_dim(dim, rankWithoutBatchDim(self, self_bdim)) + 1;
```

There are two parts to this:
- `maybe_wrap_dim(rankWithoutBatchDim(self, self_bdim))`. If dim is negative (e.g. -1), then it should be wrapped to the rank of the Tensor without the dimension being vmapped over (the batch dim).
- Then, since we moved the batch dim to the front, we need to increment the value by 1 (you end up writing dim + 1 a few times below).

It looks like there was no OpInfo with negative dimension, so it didn't catch this case (https://github.com/pytorch/pytorch/blob/dfb651452a8365a1010c63e7ace01af554ab0358/torch/testing/_internal/common_methods_invocations.py#L4197).

We should try to add a manual test for it (or update the OpInfo). The manual test will be easier for now. Example in https://github.com/pytorch/pytorch/blob/dfb651452a8365a1010c63e7ace01af554ab0358/test/functorch/test_vmap.py#L3759-L3766